openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/core/CoreMatrixAlgorithm.java

```diff
@@ -0,0 +1,538 @@
+        // Calculate all paths only inside core
+        DijkstraManyToMany algorithm = new DijkstraManyToMany(graph, chGraph, bestWeightMap, bestWeightMapCore, weighting, TraversalMode.NODE_BASED);
+
+        EdgeFilterSequence edgeFilterSequence = new EdgeFilterSequence();
+        edgeFilterSequence.add(this.additionalCoreEdgeFilter);
+
+        algorithm.setEdgeFilter(edgeFilterSequence);
+        algorithm.setTreeEntrySize(this.treeEntrySize);
+        algorithm.setHasTurnWeighting(this.hasTurnWeighting);
+        algorithm.setMaxVisitedNodes(this.maxVisitedNodes);
+        algorithm.setVisitedNodes(this.visitedNodes);
+        algorithm.setTurnWeighting(this.turnWeighting);
+        algorithm.setTargetGraphExplorer(targetGraph.createExplorer());
+        algorithm.setTargetMap(this.targetMap);
+        algorithm.setTargetSet(this.targetSet);
+        algorithm.setSwap(this.swap);
+
+        int[] entryPoints = coreEntryPoints.toArray();
+        int[] exitPoints = coreExitPoints.toArray();
+        algorithm.calcPaths(entryPoints, exitPoints);
+    }
+
+    /**
+     * /
+     * /
+     * __________UTIL
+     * /
+     * /
+     **/
+
+    private boolean isValid(MatrixLocations srcData, MatrixLocations dstData) {
+        return !(!srcData.hasValidNodes() || !dstData.hasValidNodes());
+    }
+
+    /**
+     * Search is more efficient for low source count and high destination count than the other way around.
+     * If there are more sources than destinations, they get swapped and all calculations are done backwards.
+     * The final result gets unswapped to return correct results.
+     * @param srcData original Source data
+     * @param dstData original Destination data
+     * @return
+     */
+    private boolean checkSwapSrcDst(MatrixLocations srcData, MatrixLocations dstData){
+        return(srcData.size() > dstData.size());
+    }
+
+    /**
+     * Invert the results matrix (represented by flattened array) in case src and dst were swapped
+     * @param srcData the original unswapped source data
+     * @param dstData the original unswapped destination data
+     * @param times the swapped array of results
+     * @param distances the swapped array of results
+     * @param weights the swapped array of results
+     * @return array of unswapped result arrays [times, distances, weights]
+     */
+    private float[][] swapResults(MatrixLocations srcData, MatrixLocations dstData, float[] times, float[] distances, float[] weights) {
+        boolean hasTimes = times != null;
+        boolean hasDistances = distances != null;
+        boolean hasWeights = weights != null;
+        float[] newTimes = new float[0];
+        float[] newDistances = new float[0];
+        float[] newWeights = new float[0];
+        if(hasTimes)
+            newTimes = new float[times.length];
+        if(hasDistances)
+            newDistances = new float[distances.length];
+        if(hasWeights)
+            newWeights = new float[weights.length];
```

Comment:
The arrays are initialized repeatedly with different lengths, but there is actually no need of calling the constructor twice if the length itself is made conditional like so:

```
 float[] newTimes = new float[hasTimes ? times.length : 0];
```

Reply:
done